src/main/java/reposense/wizard/OptionalPrompt.java

```diff
@@ -0,0 +1,35 @@
+package reposense.wizard;
+
+/**
+ * Represents an optional prompt that only calls the provided prompt if requested by the user.
+ */
+public class OptionalPrompt extends Prompt {
+    private final Prompt prompt;
+    private final static String YES_FLAG = "Y";
+    private final static String NO_FLAG = "N";
+    private final static String DESCRIPTION = String.format("%s/%s", YES_FLAG, NO_FLAG);
```

Comment:
Hmm for Y/N to be used as `DESCRIPTION` seems slightly odd to me. I think it may be better if we change the name of the `description` field in `Prompt` to maybe `choices`? Open to discussion for the naming.

Reply:
Considering `description` was originally meant for Y/N

Reply:
Yup sure, I named it the description to follow the field name of `Prompt`. Let me change both the field name of `Prompt` and the name of this String.

Reply:
Yea I believe that's a way better naming than what we had in the beginning